src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -4295,12 +4299,24 @@ private Expression GetRangeEnumerator(ExpressionAst condExpr)
                     Expression lhs = Compile(binaryExpr.Left);
                     Expression rhs = Compile(binaryExpr.Right);
 
-                    result = Expression.New(CachedReflectionInfo.RangeEnumerator_ctor,
-                                            lhs.Convert(typeof(int)),
-                                            rhs.Convert(typeof(int)));
+                    if ((lhs.Type == typeof(string) || lhs.Type == typeof(char))
```

Comment:
I think this will break a weird but supported scenario:
```powershell
"0".."10"
```
I know it's dumb code, but I've seen it often enough that I wouldn't break this case.

Reply:
😄 I wonder why do we have CIs if we have Jason Shirk?

While I don't know how fix this here. And I see the same problem in line 4893. I guess we can not resolve the conflict in `RangeOperator` method: "0".."9", $var1.."9", "0"..$var2 - covert string to char or int? What if user want get char array? 😕 

Another question. In line 4893 I tried to make this code faster for constants. It seems that with this addition it can becomes too complicated and duplicated. I think to leave only Expression.Call() and put all magic code in one method. Right?
I see all `Expression.Call()` has a ToDo comment ("replace this with faster code") - could you please explain the idea? Maybe we can make more right thing now?

Reply:
Assume any string with Length > 1 or if the only character is 0-9, assume int, otherwise char - that should be sufficient.

The `TODO` was written when I first wrote that code and just getting things working. I don't remember what I had in mind, if anything.

It's your call on how you want to write the code - you can avoid some duplication with a helper method if you prefer to keep the optimization.